frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx

```diff
@@ -62,14 +65,67 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
   const { handleGeocoderChanges } = useGeocoder({ formikRef });
   const parcelsService = useLayerQuery(PARCELS_LAYER_URL);
 
-  const handlePidChange = (pid: string) => {
+  /** query pims for the given pid, set data within the form if match found. Fallback to querying the parcel data layer. */
+  const handlePidChange = (pid: string, nameSpace?: string) => {
     const formattedPid = pid.replace(/-/g, '');
-    const response = parcelsService.findByPid(formattedPid);
-    handleParcelDataLayerResponse(response, dispatch);
+    if (isNaN(+formattedPid)) {
+      return;
+    }
+    fetchParcelsDetail({ pid } as any)(dispatch).then(resp => {
```

Comment:
these search functions share some common logic that I will extract out into shared functions.